lib/middleware.js

```diff
@@ -55,26 +56,33 @@ exports.expressify = function(middlewareFn) {
 
 /***
   Function: checkToken(client)
-  Desc: Middleware to process the session token -- expects a token in a request header, processes it, and
+  Desc: Middleware to process the session token or access token -- expects a token in a request header, processes it, and
         returns information about the token in the _tokendata variable on the request.
   Args: client -- client to use when talking to the user-api**/
 exports.checkToken = function (client) {
   return function(req, res, next) {
-    var sessionToken = req.headers[sessionTokenHeader];
+    var token = req.headers[sessionTokenHeader];
+    var accessTokenHeader = req.headers[authorizationHeader];
 
-    if (sessionToken == null) {
+    if (accessTokenHeader != null && 
+      accessTokenHeader.length > 7 && 
+      (accessTokenHeader.toLowerCase().indexOf('bearer ') !== -1)) {
+      token = accessTokenHeader.split(' ')[1];
+    }
+
+    if (token == null) {
```

Comment:
Not sure if this is standard across all our Node repos, but I see `blip` and `chrome-uploader` don't do the comparison against `null`, so just `if (!token) {`.

Also,  we could also check against an empty string, so `if (!token || token == "") {`.